include/internal/rcu.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_RCU_H
+# define OPENSSL_RCU_H
+# pragma once
+
+typedef void (*rcu_cb_fn)(void *data);
+
+typedef struct rcu_lock_st *CRYPTO_RCU_LOCK;
+
+CRYPTO_RCU_LOCK ossl_rcu_lock_new(int num_writers);
+void ossl_rcu_lock_free(CRYPTO_RCU_LOCK lock);
+void ossl_rcu_read_lock(CRYPTO_RCU_LOCK lock);
+void ossl_rcu_write_lock(CRYPTO_RCU_LOCK lock);
+void ossl_rcu_write_unlock(CRYPTO_RCU_LOCK lock);
+void ossl_rcu_read_unlock(CRYPTO_RCU_LOCK lock);
+void ossl_synchronize_rcu(CRYPTO_RCU_LOCK lock);
+int ossl_rcu_call(CRYPTO_RCU_LOCK lock, rcu_cb_fn cb, void *data);
```

Comment:
I am sorry for nitpicking, but we never use  pointer typedefs elsewhere in the code. They might be pretty confusing when used in code sometimes.  Could we change it so the the CRYPTO_RCU_LOCK typedef is just for the struct and not a pointer to a struct?

Reply:
I would also prefer this, it's more consistent with how we've done things.

Reply:
yeah, ok